pkg/k8splan/watcher.go

```diff
@@ -221,6 +307,27 @@ func (w *watcher) start(ctx context.Context) {
 	}
 }
 
+func (w *watcher) updateSecret(core corecontrollers.Interface, secret *v1.Secret) (*v1.Secret, error) {
+	var resultingSecret *v1.Secret
+	err := retry.OnError(retry.DefaultBackoff,
+		func(err error) bool {
+			if apierrors.IsConflict(err) {
+				return false
+			}
+			return true
+		},
+		func() error {
+			var err error
+			resultingSecret, err = core.Secret().Update(secret)
+			return err
+		})
+	if err == nil {
```

Comment:
Can we refactor this so it looks for an error and actos accordily and returns nil for the resultingSecret and nil for the err at the end of the func?

Reply:
I am working to retain the "standard" behavior which is to always return what is resulting from `core.Secret().Update()`, but we only want to update the last applied resource version in the event that the secret update was successful i.e. where `err == nil`